src/alloc_pool.c

```diff
@@ -435,7 +437,8 @@ void qd_dealloc(qd_alloc_type_desc_t *desc, qd_alloc_pool_t **tpool, char *p)
     assert (desc->header  == PATTERN_FRONT);
     assert (desc->trailer == PATTERN_BACK);
     assert (item->header  == PATTERN_FRONT);
-    assert (*((uint32_t*) (p + desc->total_size)) == PATTERN_BACK);
+    const uint32_t pb = PATTERN_BACK;
```

Comment:
```
/home/travis/build/apache/qpid-dispatch/src/alloc_pool.c:440:20: error: unused variable 'pb' [-Werror,-Wunused-variable]

    const uint32_t pb = PATTERN_BACK;
```

Reply:
(That's because the next line is an assert, so it may not be always compiled in...)